### PR TITLE
Add alias_name_source to r/d/kubernetes_auth_backend_role

### LINKED ISSUE #1327: vault_kubernetes_auth_backend_role should support alias_name_source
## Terraform Version
$ terraform -v
Terraform v1.1.3
on darwin_amd64

Affected Resource(s)
Please list the resources as a list, for example:

`vault_kubernetes_auth_backend_role`

Terraform Configuration Files
```
resource "vault_kubernetes_auth_backend_role" "role" {
  backend                          = "kubernetes/cluster1"
  role_name                        = "kube_cluster1_team1_app1"
  bound_service_account_names  = "app1-svc"
  bound_service_account_namespaces = "team1"
  token_policies                   = ["default"]
  #alias_name_source  = "serviceaccount_name" # default is 'serviceaccount_uid' but this attribute is not supported either way
}
```

## Expected Behavior
This element is exposed in the GUI for the kubernetes role once created via terraform. The GUI tooltip for the field is:
```
Source to use when deriving the Alias name. valid choices: "serviceaccount_uid" : <token.uid> e.g. 474b11b5-0f20-4f9d-8ca5-65715ab325e0 (most secure choice) "serviceaccount_name" : <namespace>/<serviceaccount> e.g. vault/vault-agent default: "serviceaccount_uid"
```

Terraform should allow me to set the alias_name_source upon role creation.

## Actual Behavior
Terraform will create kubernetes auth roles will always get serviceaccount_uid as the naming convention for newly generated entity aliases the role generates. As the serviceaccount_uids are only known at runtime from the kubernetes deployment itself, we can not effectively predict these alias names which means only token bound policies can be used to manage these roles.

## References
I was unable to find any other issues referencing this.

### PR DESCRIPTION
**Pull request by benashz**
This PR adds support for configuring `alias_name_source` in the `kubernetes_auth_backend_role` resource. It also provides the same field as an attribute of the corresponding datasource.

### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" comments, they generate extra noise for pull request followers and do not help prioritize the request

Closes #1327 

Release note for [CHANGELOG](https://github.com/hashicorp/terraform-provider-vault/blob/master/CHANGELOG.md):

```release-note

```

Output from acceptance testing:

```
$ make testacc TESTARGS='-v -test.run TestAccKubernetesAuthBackendRole'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test -v -v -test.run TestAccKubernetesAuthBackendRole -timeout 30m ./...

[...]

ok      github.com/hashicorp/terraform-provider-vault/util      (cached) [no tests to run]
=== RUN   TestAccKubernetesAuthBackendRoleDataSource_basic
--- PASS: TestAccKubernetesAuthBackendRoleDataSource_basic (1.85s)
=== RUN   TestAccKubernetesAuthBackendRoleDataSource_full
--- PASS: TestAccKubernetesAuthBackendRoleDataSource_full (1.81s)
=== RUN   TestAccKubernetesAuthBackendRole_import
--- PASS: TestAccKubernetesAuthBackendRole_import (2.20s)
=== RUN   TestAccKubernetesAuthBackendRole_basic
--- PASS: TestAccKubernetesAuthBackendRole_basic (1.81s)
=== RUN   TestAccKubernetesAuthBackendRole_update
--- PASS: TestAccKubernetesAuthBackendRole_update (3.12s)
=== RUN   TestAccKubernetesAuthBackendRole_full
--- PASS: TestAccKubernetesAuthBackendRole_full (1.79s)
=== RUN   TestAccKubernetesAuthBackendRole_fullUpdate
--- PASS: TestAccKubernetesAuthBackendRole_fullUpdate (7.03s)
PASS
ok      github.com/hashicorp/terraform-provider-vault/vault     20.320s
```

### CODE REVIEW

File: website/docs/r/kubernetes_auth_backend_role.html.md [hunk 1]
```diff
@@ -44,11 +44,16 @@ The following arguments are supported:
...
+implications to be aware of.
+ 
+* `alias_name_source` - (Optional, default: `serviceaccount_uid`) Configures how identity aliases are generated.
+   Valid choices are: `serviceaccount_uid`, `serviceaccount_name`. (vault-1.9+)
+
 ### Common Token Arguments
 
 These arguments are common across several Authentication Token resources since Vault 1.2.
-
-* `token_ttl` - (Optional) The incremental lifetime for generated tokens in number of seconds.
```
Comment (vinay-gopalan) [inline website/docs/r/kubernetes_auth_backend_role.html.md:51]: Has this since been deprecated?

Comment (benashz) [reply website/docs/r/kubernetes_auth_backend_role.html.md:51]: I think `token_ttl` replaced `ttl` which has now been removed.

Comment (benashz) [reply website/docs/r/kubernetes_auth_backend_role.html.md:51]: We could probably go through and remove all of the chatter about `Vault 1.2...`

Comment (vinay-gopalan) [reply website/docs/r/kubernetes_auth_backend_role.html.md:51]: Cool no problem, just wanted to confirm!

File: vault/data_source_kubernetes_auth_backend_role_test.go [hunk 2]
```diff
@@ -98,7 +97,8 @@ func TestAccKubernetesAuthBackendRoleDataSource_full(t *testing.T) {
 		CheckDestroy: testAccCheckKubernetesAuthBackendConfigDestroy,
 		Steps: []resource.TestStep{
 			{
-				Config: testAccKubernetesAuthBackendRoleConfig_full(backend, role, ttl, maxTTL, audience),
+				Config: testAccKubernetesAuthBackendRoleDataSourceConfig_full(
+					backend, role, "serviceaccount_name", ttl, maxTTL, audience),
```
Comment (vinay-gopalan) [inline vault/data_source_kubernetes_auth_backend_role_test.go:101]: Are we missing a check for `alias_name_source == "serviceaccount_name"` here?

**Review by vinay-gopalan — APPROVED** [review]
Looks good! Have a question about a potential missing check

File: vault/data_source_kubernetes_auth_backend_role_test.go (hunk 2 above)
Comment (benashz) [reply vault/data_source_kubernetes_auth_backend_role_test.go:101]: Good eye! Must have gotten dropped...

Comment (benashz) [reply vault/data_source_kubernetes_auth_backend_role_test.go:101]: Fixed in https://github.com/hashicorp/terraform-provider-vault/pull/1336/commits/a8b5fde9f877fe22d0d5492f74054573df6275e9

**Pull request merged**